scenario/utils.ts

```diff
@@ -254,3 +266,183 @@ export async function fetchLogs(
     return contract.queryFilter(filter, fromBlock, toBlock);
   }
 }
+
+export async function setNextBaseFeeToZero(dm: DeploymentManager) {
+  await dm.hre.network.provider.send('hardhat_setNextBlockBaseFeePerGas', ['0x0']);
+}
+
+export async function mineBlocks(dm: DeploymentManager, blocks: number) {
+  await dm.hre.network.provider.send('hardhat_mine', [`0x${blocks.toString(16)}`]);
+}
+
+export async function setNextBlockTimestamp(dm: DeploymentManager, timestamp: number) {
+  await dm.hre.ethers.provider.send('evm_setNextBlockTimestamp', [timestamp]);
+}
+
+export async function executeOpenProposal(
+  dm: DeploymentManager,
+  { id, startBlock, endBlock }: OpenProposal
+) {
+  const governor = await dm.getContractOrThrow('governor');
+  const blockNow = await dm.hre.ethers.provider.getBlockNumber();
+  const blocksUntilStart = startBlock - blockNow;
+  const blocksUntilEnd = endBlock - Math.max(startBlock, blockNow);
+
+  if (blocksUntilStart > 0) {
+    await mineBlocks(dm, blocksUntilStart);
+  }
+
+  const compWhales = dm.network === 'mainnet' ? COMP_WHALES.mainnet : COMP_WHALES.testnet;
+
+  if (blocksUntilEnd > 0) {
+    for (const whale of compWhales) {
+      try {
+        // Voting can fail if voter has already voted
+        const voter = await impersonateAddress(dm, whale);
+        await setNextBaseFeeToZero(dm);
+        await governor.connect(voter).castVote(id, 1, { gasPrice: 0 });
+      } catch (err) {
+        debug(`Error while voting for ${whale}`, err.message);
+      }
+    }
+    await mineBlocks(dm, blocksUntilEnd);
+  }
+
+  // Queue proposal (maybe)
+  const state = await governor.state(id);
+  if (state == ProposalState.Succeeded) {
+    await setNextBaseFeeToZero(dm);
+    await governor.queue(id, { gasPrice: 0 });
+  }
+
+  const proposal = await governor.proposals(id);
+  await setNextBlockTimestamp(dm, proposal.eta.toNumber() + 1);
+
+  // Execute proposal (w/ gas limit so we see if exec reverts, not a gas estimation error)
+  await setNextBaseFeeToZero(dm);
+  await governor.execute(id, { gasPrice: 0, gasLimit: 12000000 });
+}
+
+// Instantly executes some actions through the governance proposal process
+export async function fastGovernanceExecute(
+  dm: DeploymentManager,
+  proposer: SignerWithAddress,
+  targets: string[],
+  values: BigNumberish[],
+  signatures: string[],
+  calldatas: string[]
+) {
+  const governor = await dm.getContractOrThrow('governor');
+
+  await setNextBaseFeeToZero(dm);
+
+  const proposeTxn = await (
+    await governor.connect(proposer).propose(
+      targets,
+      values,
+      signatures,
+      calldatas,
+      'FastExecuteProposal',
+      { gasPrice: 0 }
+    )
+  ).wait();
+  const proposeEvent = proposeTxn.events.find(event => event.event === 'ProposalCreated');
+  const [id, , , , , , startBlock, endBlock] = proposeEvent.args;
+
+  await executeOpenProposal(dm, { id, startBlock, endBlock });
+}
+
+async function relayMumbaiMessage(
```

Comment:
I wonder if we should have files for each set of bridge-specific stuff 🤔

Reply:
Yeah that would be a cleaner way to organize things, since each `relayXMessage` is going to be pretty large.

Reply:
right now, moving `relayMumbaiMessage` out of `utils.ts` creates a circular dependency with some of the helper functions.

i'll move some of these things around. going to do it in a follow-up PR though